leakcanary-android-core/src/main/java/leakcanary/ConfigBuilder.kt

```diff
@@ -0,0 +1,90 @@
+package leakcanary
+
+import shark.LeakingObjectFinder
+import shark.MetadataExtractor
+import shark.ObjectInspector
+import shark.ReferenceMatcher
+
+/**
+ * Builder for [LeakCanary.Config] intended to use only by Java callers.
+ *
+ * Usage:
+ * ```
+ * LeakCanary.Config config = LeakCanary.getConfig().newBuilder()
+ *    .dumpHeap(false)
+ *    .retainedVisibleThreshold(3)
+ *    .maxStoredHeapDumps(10)
+ *    .build();
+ * LeakCanary.setConfig(config);
+ * ```
+ *
+ * Kotlin callers should be using `copy()` method instead:
+ * ```
+ * LeakCanary.config = LeakCanary.config.copy(dumpHeap = false)
+ * ```
+ */
+@Suppress("TooManyFunctions")
+class ConfigBuilder(config: LeakCanary.Config) {
```

Comment:
I wanted to hide this class from Kotlin users as well, but then the `LeakCanary.config.newBuilder()` won't work as it won't be able to find this class. Choosing between the two, I would rather have `newBuilder()` method hidden

Reply:
> hide this class from Kotlin users as well

Did you hide it by using the "SinceKotlin(999)" hack?

> but then the LeakCanary.config.newBuilder() won't work

What does "won't work" mean here?

Reply:
I did hide it with `SinceKotlin("999")`. `LeakCanary.config.newBuilder()` can't see the `ConfigBuilder` because it is hidden from the Kotlin users with aforementioned hack 😞